birdhouse/components/weaver/docker-compose-extra.yml

```diff
@@ -42,38 +42,18 @@ services:
     #   Weaver is started, they will all be able to request them and register the providers correctly.
     #   This is needed because simply adding 'depends_on' only ensures that containers are 'running', but startup of
     #   WPS applications themselves are not necessarily completed. Successful HTTP responses ensure they are 'ready'.
-    #image: pavics/weaver:${WEAVER_VERSION}-manager
-    image: pavics/weaver:${WEAVER_VERSION}-wait-wps
-    build:
-      context: ./components/weaver
-      dockerfile: Dockerfile
-      args:
-        WEAVER_VERSION: ${WEAVER_VERSION}
+    image: pavics/weaver:${WEAVER_VERSION}-manager
     ports:
       - "4001:4001"
     environment:
       HOSTNAME: ${PAVICS_FQDN}
       FORWARDED_ALLOW_IPS: "*"
     #env_file:
     #  - ./config/mongodb/credentials.env
-    # ensure other referenced services are created beforehand to avoid booting timing issues
     depends_on:
-      - catalog
-      - finch
-      - flyingpigeon
-      - hummingbird
-      - malleefowl
-      - raven
-      #- wait-for-wps
       - mongodb
     links:
       - mongodb
-      - catalog
-      - finch
-      - flyingpigeon
-      - hummingbird
-      - malleefowl
-      - raven
```

Comment:
The links where there originally only because the `dockerize -wait` calls was using them. 

With the new registration `post-docker-compose-up` method, we have 2 possibilities, each with different concerns to consider: 

1. If we register like I it did for now in the script, using the public endpoint, we make sure that even if Magpie permissions over Weaver-converted WPS providers & their sub-processes were not correctly set, the user cannot obtain indirect access to them via Weaver, since Weaver querying them will cause Forbidden access. 
The disadvantage though is that we redo the request from the top, adding some (relatively minimal) delay to resolve the path + the extra permission resolution rather then requesting directly.

2. If we instead send requests directly via docker-link references, we avoid the full loop back through the server entrypoint, gaining some speed, but at risk of ignoring permissions. The configuration parameter `WEAVER_WPS_PROVIDERS` will also need to be somewhat more complicated as the internal URLs will have to be provided explicitly.

Reply:
I have a slight preference for option 2.  As for the prob of "ignoring permissions", well the code use the magpie admin credential to hit the various WPS so no permissions will be enforced anyway, I see no difference.

However, if I understand how Weaver is being deployed, it is still behind Magpie so Magpie can still apply the same WPS permissions even if the WPS is behind Weaver instead of behind Twitcher directly.  I guess Magpie will have to be Weaver aware so some modifications on Magpie side.

Reply:
The resulting registered providers will be problematic.
If I register them as let's say `http://finch:8095` to call directly, requests such as 
```
https://<PAVICS_FQDN>/twitcher/ows/proxy/weaver/providers/finch/processes/ice_days
```
(aka OGC-API REST equivalent of WPS DescribeProcess) will call 
```
http://finch:8095?service=wps&request=DescribeProcess&identifier=ice_days&version=1.0.0
```
instead of 
```
https://<PAVICS_FQDN>/twitcher/ows/proxy/finch?service=wps&request=DescribeProcess&identifier=ice_days&version=1.0.0
```

This means if a permission was set to block this `DescribeProcess` request on the real finch WPS side, but not on Weaver, a user could bypass the permissions and run the process directly via Weaver. 
The process is not executed as an admin, but as the submitting user. 

Weaver would always need to be kept in sync with other services it registers. 
This is something I definitely don't plan to handle on Weaver nor Magpie side, since it is neither their job.
Maybe Cowbird could handle that at a later time.

Until then, by using the `/twitcher/ows/proxy` route, even if we forget by mistake to update the API permission for Weaver, the corresponding operation will still respect the WPS permissions applied for the user requesting it.

Reply:
Opps sorry, I mixed up the "wait" and the "register" step.  Totally agree the register step should use the public URL. "wait" step may or may not use the public URL.

> Until then, by using the /twitcher/ows/proxy route, even if we forget by mistake to update the API permission for Weaver, the corresponding operation will still respect the WPS permissions applied for the user requesting it.

Oh nice, then do we even need to set any specific Weaver permissions?  Since this fallback will "catch all" anyways?

Reply:
It would be useful for early rejection of the requests. 
For example, Weaver execute would create a job, dispatch it to the WPS, collect logs and such, just to indicate permission denied. 
But no issue in being double secure about it.

Reply:
> It would be useful for early rejection of the requests.
> For example, Weaver execute would create a job, dispatch it to the WPS, collect logs and such, just to indicate permission denied.

Ah, understood now.

> But no issue in being double secure about it.

Indeed.